subsys/bluetooth/host/conn.c

```diff
@@ -2675,6 +2675,7 @@ int bt_conn_le_create(const bt_addr_le_t *peer,
 
 #if defined(CONFIG_BT_SMP)
 	if (IS_ENABLED(CONFIG_BT_PRIVACY) &&
+	    (bt_dev.le.rl_entries > 0) &&
 	    (!bt_dev.le.rl_size || bt_dev.le.rl_entries > bt_dev.le.rl_size)) {
```

Comment:
This can be (purely mathematically) simplified.
```suggestion
	    (bt_dev.le.rl_entries > bt_dev.le.rl_size)) {
```